File: solutions/zig/leap/1/leap.zig
pub fn isLeapYear(year: u32) bool {
    const by400 = (year % 400) == 0;
    const by4 = (year % 4) == 0;
    const by100 = (year % 100) == 0;
    return by400 or (by4 and !by100);
}

File: solutions/lua/leap/1/leap.lua
local leap_year = function(number)
	local by400 = (number % 400) == 0
	local by4 = (number % 4) == 0
	local by100 = (number % 100) == 0
	return by400 or (by4 and not by100)
end

return leap_year
